src/Core/System/User/UserEntity.php

```diff
@@ -401,7 +278,7 @@ public function setAdmin(bool $admin): void
         $this->admin = $admin;
     }
 
-    public function getAclRoles(): AclRoleCollection
+    public function getAclRoles(): ?AclRoleCollection
```

Comment:
Is this not a break since this changes the behavior?

Reply:
Good point. I considered adding a condition to return an empty AclRoleCollection if the class variable is null, but that wouldn’t be ideal. An empty collection would suggest that the ACL role associations were loaded but empty.
A null value clearly indicates that the association was not loaded.

I think we should keep your change because, as you mentioned, every caller has already been responsible for ensuring a valid collection is returned.

Works for me! 👍